scipy/interpolate/tests/test_rgi.py

```diff
@@ -939,3 +939,15 @@ def test_invalid_xi_dimensions(self):
                "RegularGridInterpolator has dimension 1")
         with assert_raises(ValueError, match=msg):
             interpn(points, values, xi)
+
+    def test_readonly_grid(self):
+        # https://github.com/scipy/scipy/issues/17716
+        x = np.linspace(0, 4, 5)
+        y = np.linspace(0, 5, 6)
+        z = np.linspace(0, 6, 7)
+        points = (x, y, z)
+        for d in points:
+            d.flags.writeable = False
+        values = np.ones((5, 6, 7))
+        point = np.array([2.21, 3.12, 1.15])
+        interpn(points, values, point)
```

Comment:
Minor: if you're adding more fixes/tests, it'd be nice to make the test exercise `RegularGridInterpolator(points, values)(point)`. We had an issue with things slipping because tests only exercised interpn not RGI.

Reply:
Thanks! I added the suggestion to the test.